tests/std/tests/P0784R7_library_support_for_more_constexpr_containers/test.cpp

```diff
@@ -272,8 +276,152 @@ constexpr void test_compiletime() {
+        assert(a.object.value == 0);
+        std::allocator_traits<Alloc<A<int>>>::destroy(alloc, &a.object);
+
+        assert(std::allocator_traits<Alloc<A<int>>>::select_on_container_copy_construction(alloc).id == 11);
+
+        assert(std::allocator_traits<Alloc<A<int>>>::max_size(alloc)
+               == std::numeric_limits<Alloc<A<int>>::size_type>::max() / sizeof(Alloc<A<int>>::value_type));
+    }
+    {
+        storage_for<nontrivial_A<int>> a;
+        Alloc<nontrivial_A<int>> alloc{10};
+        assert(alloc.id == 10);
+
+        auto result = std::allocator_traits<Alloc<nontrivial_A<int>>>::allocate(alloc, 10);
+        assert(result != nullptr);
+        std::allocator_traits<Alloc<nontrivial_A<int>>>::deallocate(alloc, result, 10);
+
+        std::allocator_traits<Alloc<nontrivial_A<int>>>::construct(alloc, &a.object, 10);
+        assert(a.object.value == 10);
+        std::allocator_traits<Alloc<nontrivial_A<int>>>::destroy(alloc, &a.object);
+
+        assert(std::allocator_traits<Alloc<nontrivial_A<int>>>::select_on_container_copy_construction(alloc).id == 11);
+
+        assert(std::allocator_traits<Alloc<nontrivial_A<int>>>::max_size(alloc)
+               == std::numeric_limits<Alloc<nontrivial_A<int>>::size_type>::max()
+                      / sizeof(Alloc<nontrivial_A<int>>::value_type));
+    }
+    {
+        storage_for<nontrivial_A<int>> a;
+        Alloc<nontrivial_A<int>, true> alloc{10};
+
+        std::allocator_traits<Alloc<nontrivial_A<int>, true>>::construct(alloc, &a.object, 10);
+        assert(a.object.value == 10);
+        std::allocator_traits<Alloc<nontrivial_A<int>, true>>::destroy(alloc, &a.object);
+    }
+    {
+        storage_for<nontrivial_A<int>> a;
+        Alloc<nontrivial_A<int>, false, true> alloc{10};
+
+        std::allocator_traits<Alloc<nontrivial_A<int>, false, true>>::construct(alloc, &a.object, 10);
+        assert(a.object.value == 10);
+        std::allocator_traits<Alloc<nontrivial_A<int>, false, true>>::destroy(alloc, &a.object);
+    }
+    {
+        storage_for<nontrivial_A<int>> a;
+        Alloc<nontrivial_A<int>, true, true> alloc{10};
+
+        std::allocator_traits<Alloc<nontrivial_A<int>, true, true>>::construct(alloc, &a.object, 10);
+        assert(a.object.value == 10);
+        std::allocator_traits<Alloc<nontrivial_A<int>, true, true>>::destroy(alloc, &a.object);
+    }
+}
+static_assert((test_compiletime_allocator_traits(), true));
+
+constexpr void test_compiletime_allocator() {
+    {
+        auto result = std::allocator<A<int>>{}.allocate(10);
+        std::allocator<A<int>>{}.deallocate(result, 10);
+    }
+    {
+        auto result = std::allocator<nontrivial_A<int>>{}.allocate(10);
+        std::allocator<nontrivial_A<int>>{}.deallocate(result, 10);
+    }
+}
+static_assert((test_compiletime_allocator(), true));
+#endif // defined(__clang__ ) || defined(__EDG__)
 #endif // _HAS_CXX20 && defined(__cpp_constexpr_dynamic_alloc)
 
+#if _HAS_CXX20

```

Comment:
We don't need any switches for `_HAS_CXX20` because P0784 is C++20-specific and this test is run with `usual_latest_matrix` which always implies C++20 (or beyond) mode!  There are some `_HAS_CXX20`s pre-existing in this file, so please audit the entire test to remove these.